lib/curvature/collector.py

```diff
@@ -204,7 +204,10 @@ def join_ways(self):
             sys.stderr.flush()
 
         for route, route_data in self.routes.iteritems():
-            ways = route_data['ways']
+            # Sort ways by OSM id so that joining always happens in the same order
+            # even if the parser returns them in a different order.
+            ways = sorted(route_data['ways'], key=lambda way: way['id'])
```

Comment:
Nope, all this does is set up the starting order of the ways before they are joined. The loop below will start with the one with the lowest id as the base, then cycle through the rest of the ways trying to join them on. Each time a way is added to the joined group the rest will be run through again to see if they can be added onto the now longer collection.

All that ordering them does is to ensure that process uses a consistent order for join attempts -- the resulting collection will still be in connection-order.